pkg/snet/network.go

```diff
@@ -263,18 +279,24 @@ func (c Conn) RemotePort() uint16 { return c.rPort }
 // Network returns network of connection.
 func (c Conn) Network() string { return c.network }
 
-func disassembleAddr(addr net.Addr) (pk cipher.PubKey, port uint16) {
+func disassembleAddr(addr net.Addr) (pk cipher.PubKey, port uint16, retErr error) {
 	strs := strings.Split(addr.String(), ":")
 	if len(strs) != 2 {
-		panic(fmt.Errorf("network.disassembleAddr: %v %s", "invalid addr", addr.String()))
```

Comment:
I feel like panics here are appropriate, because is the `addr` cannot be disassembled to `pk:port`, it signals an invalid `snet.Conn`.

Reply:
This may happen if one of our subsystems implements `net.Addr` the wrong way, the way we consider wrong. So, I'd stick to panic in this particular case, cause it will raise only on first test runs and will help us identify error fast instead of looking through logs

Reply:
@evanlinjin @Darkren Thanks for your opinions. I agree.